services/ansibler/server/loadbalancers.go

```diff
@@ -40,106 +40,316 @@ const (
-					directory := filepath.Join(k8sDirectory, fmt.Sprintf("%s-%s", lb.LbCluster.ClusterInfo.Name, lb.LbCluster.ClusterInfo.Hash))
+		k8sClusterName := k8sClusterName
+		lbInfo := lbInfo
+
+		//set up all lbs for the k8s cluster, since single k8s can have multiple LBs
+		k8sDirectory := filepath.Join(baseDirectory, outputDirectory, fmt.Sprintf("%s-%s-lbs", k8sClusterName, utils.CreateHash(4)))
+
+		//generate inventory for all LBs with k8s nodes
+		if err := generateK8sBaseFiles(k8sDirectory, lbInfo); err != nil {
+			log.Error().Msgf("error while generating base directory for %s : %v", k8sClusterName, err)
+			//continue with the next k8s cluster LBs
+			continue
+		}
+
+		//process LB clusters for single K8s cluster
+		errGroupK8s.Go(func() error {
+			var errGroupLB errgroup.Group
+			var apiServerLB *LBData
+
+			//iterate over all LBs for a single k8s cluster
+			for _, lb := range lbInfo.LbClusters {
+				lb := lb
+				directory := filepath.Join(k8sDirectory, fmt.Sprintf("%s-%s", lb.DesiredLbCluster.ClusterInfo.Name, lb.DesiredLbCluster.ClusterInfo.Hash))
+
+				errGroupLB.Go(func() error {
 					log.Info().Msgf("Setting up the LB %s", directory)
-					func(directory, k8sDirectory string, lb *LBData) {
-						//set up the individual LB
-						errGroupLB.Go(func() error {
-							//set up nginx
-							err := setUpNginx(lb.LbCluster, lbInfo.TargetK8sNodepool, directory)
-							if err != nil {
-								return fmt.Errorf("error while setting up the nginx %s : %w", lb.LbCluster.ClusterInfo.Name, err)
-							}
-							//check for DNS change
-							if utils.ChangedAPIEndpoint(lb.CurrentDNS, lb.LbCluster.Dns) {
-								err = changedAPIEp(lb, k8sDirectory)
-								if err != nil {
-									return fmt.Errorf("error while changing the endpoint for %s : %w", lb.LbCluster.ClusterInfo.Name, err)
-								}
-							}
-							return nil
-						})
-					}(directory, k8sDirectory, lb)
-				}
-				err := errGroupLB.Wait()
-				if err != nil {
-					return fmt.Errorf("error while setting up the loadbalancers for k8s cluster %s : %w", k8sClusterName, err)
+					return setUpNginx(lb.DesiredLbCluster, lbInfo.TargetK8sNodepool, directory)
+				})
+
+				for _, role := range lb.DesiredLbCluster.Roles {
+					if role.RoleType == pb.RoleType_ApiServer {
+						apiServerLB = lb
+					}
 				}
-				//Clean up
-				if err := os.RemoveAll(k8sDirectory); err != nil {
-					return fmt.Errorf("error while deleting files: %w", err)
+			}
+
+			if err := errGroupLB.Wait(); err != nil {
+				return fmt.Errorf("error while setting up the loadbalancers for k8s cluster %s : %w", k8sClusterName, err)
+			}
+
+			if err := HandleAPIEndpointChange(apiServerLB, lbInfo, k8sDirectory); err != nil {
+				return fmt.Errorf("failed to find a candidate for the Api Server: %w", err)
+			}
+
+			return os.RemoveAll(k8sDirectory)
+		})
+	}
+
+	return errGroupK8s.Wait()
+}
+
+func HandleAPIEndpointChange(apiServer *LBData, k8sCluster *LBInfo, k8sDirectory string) error {
```

Comment:
This function should probably not be exported ~`HandleAPIEndpointChange`~ -> `handleAPIEndpointChange`